tiledb/sm/query/sparse_global_order_reader.cc

```diff
@@ -194,22 +194,24 @@ Status SparseGlobalOrderReader::dowork() {
     return Status::Ok();
   }
 
-  // TODO Get rid of the copy here by keeping an end index in resultbase.
-  // Try to make a result cell slab that will fit in the user's buffers, at
-  // least for fixed sized attributes. Since the copy code might modify the
-  // result cell slabs, this is also good to have a copy.
+  // Compute an initial boundary for the copy.
   auto it = read_state_.result_cell_slabs_.begin();
-  std::vector<ResultCellSlab> to_copy;
-  while (num_cells && it != read_state_.result_cell_slabs_.end()) {
-    auto tile = it->tile_;
+  copy_end_.first = 0;
+  while (it != read_state_.result_cell_slabs_.end()) {
     if (it->length_ > num_cells) {
-      to_copy.emplace_back(tile, it->start_, num_cells);
+      copy_end_.first++;
+      copy_end_.second = it->start_ + num_cells;
+      break;
```

Comment:
Why is this here?

Reply:
The previous code checked num_cells in the while condition to break when there is no more room to copy, which wasn't accurate since is could also go below 0. I removed the num_cells condition and added the break.